### PR TITLE
New adapter: Screencore

### PR DESCRIPTION
**Pull request by screencore-io**
Link to doc - https://github.com/prebid/prebid.github.io/pull/4760

### CODE REVIEW

File: static/bidder-info/screencore.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+endpoint: 'http://h1.screencore.io/?kp={{.AccountID}}&kn={{.SourceId}}'
```
Comment (onkarvhanumante) [inline static/bidder-info/screencore.yaml:1]: endpoint mentioned is reachable

```
oh0387@oh0387s-MacBook-Pro ~ % curl -i http://h1.screencore.io/
HTTP/1.1 200 OK
Server: nginx
Date: Wed, 26 Jul 2023 06:34:30 GMT
Content-Type: application/json; charset=UTF-8
Content-Length: 10
Connection: keep-alive
Access-Control-Allow-Origin: *
```

File: adapters/screencore/screencore.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+		if len(request.Device.IP) > 0 {
+			headers.Add("X-Forwarded-For", request.Device.IP)
+		}
+	}
+
+	return headers
+}
+
+func (a *adapter) MakeRequests(openRTBRequest *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) (requestsToBidder []*adapters.RequestData, errs []error) {
+	screencoreExt, err := getImpressionExt(&openRTBRequest.Imp[0])
```
Comment (onkarvhanumante) [inline adapters/screencore/screencore.go:57]: couple of points here:

- should add length check before accessing 0th index element from `openRTBRequest.Imp`
- Only 0th index imp is being consider. Does this implies Screencore will only bid on first impression? or do bidder server expects all imp to have some params in `imp.ext`

Comment (onkarvhanumante) [inline adapters/screencore/screencore.go:57]: Only 0th index imp is being consider. Does this implies Screencore will only bid on first impression? or do bidder server expects all imp to have some params in `imp.ext`

File: adapters/screencore/screencore.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	openRTBRequest.Imp[0].Ext = nil
```
Comment (onkarvhanumante) [inline adapters/screencore/screencore.go:62]: any specific reason to set `Ext` to nil?

Comment (screencore-io) [reply adapters/screencore/screencore.go:62]: we will take ext values from uri -  http://h1.screencore.io/?kp={{.AccountID}}&kn={{.SourceId}}. So it's unnecessary to add ext in request

File: adapters/screencore/screencore.go (hunk 2 above)
Comment (screencore-io) [reply adapters/screencore/screencore.go:57]: Sure, mentioned it [here](https://github.com/prebid/prebid.github.io/pull/4760/files#diff-f7b091a53dfe436d34ee903c9045d5cfc07ee38e2ba27e6c88f59500f4bbfca7R27)

File: static/bidder-info/screencore.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+maintainer:
+  email: 'connect@screencore.io'
```
Comment (Sonali-More-Xandr) [inline static/bidder-info/screencore.yaml:3]: @screencore-io 

prebid team has sent verification mail on connect@screencore.io . Please respond back with "received" message to complete verification

File: adapters/screencore/screencore.go (hunk 2 above)
Comment (screencore-io) [reply adapters/screencore/screencore.go:63]: fixed

**Review by onkarvhanumante — APPROVED** [review]

**Review by Sonali-More-Xandr — APPROVED** [review]

**Pull request merged**